Refine DatePicker API and impement DatePickerStyle support [Gtk][Mac]

sevoku opened the pull request:
This PR adds DatePickerStyle support to Gtk and Mac backends, and includes many fixes in the backend implementations.

* GTK:
  * fix component selection
  * fix (keep) seconds when other component chages
  * fix keyboard input (incl. arrow keys and separators)
  * fix mouse selection
  * add internationalization support
    * use localized separators
    * use localized component order
    * no 12/24 support
  * add DatePickerStyle support (show only relevant components)
  * add clipboard support
* MAC:
  * add DatePickerStyle support
  * fix DatePicker timezone conversion

For WPF we still need a useful DateTimePicker component. Maybe something external from NuGet?

slluis commented on Xwt.Gtk/Xwt.GtkBackend/DatePickerBackend.cs:
```diff
@@ -85,253 +96,348 @@ void HandleValueChanged (object sender, EventArgs e)
+						" " + "HH" + timeSeparator + "mm" + timeSeparator + "ss";
+
+					if (styleFormats.ContainsKey (value))
+						format = styleFormats [value];
```
TryGetValue should be used here

slluis commented on Xwt.Gtk/Xwt.GtkBackend/DatePickerBackend.cs:
```diff
@@ -85,253 +96,348 @@ void HandleValueChanged (object sender, EventArgs e)
+			                                                          DateTime.MaxValue.Ticks,
+			                                                          TimeSpan.TicksPerSecond)
 			{
+				styleFormats[DatePickerStyle.Date] = CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern;
```
Shouldn't styleFormats be static?

slluis commented on Xwt.Gtk/Xwt.GtkBackend/DatePickerBackend.cs:
```diff
@@ -85,253 +96,348 @@ void HandleValueChanged (object sender, EventArgs e)
+				if (styleFormats.ContainsKey (Style))
+					Text = dateTime.ToString (styleFormats [Style]);
+				else
+					Text = CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern
```
We don't need to support this scenario. The style should be a valid enum value.

sevoku commented:
You're right about the formats in Gtk. I've made them static and did some cleanup.

sevoku commented:
I've added an other timezone fix on Mac, removed the static dictionary from the Gtk backend and renamed the Min/Max values (consistency). Its ok to merge now.
